File: Algorithms/Insertion Sort/InsertionSort.kt
fun insertionSort(array: IntArray) {
    val n = array.size
    for (j in 1 until n) {
        val key = array[j]
        var i = j - 1
        while (i > -1 && array[i] > key) {
            array[i + 1] = array[i]
            i--
        }
        array[i + 1] = key
    }
}


fun main() {
    val arr1 = intArrayOf(9, 14, 3, 2, 43, 11, 58, 22)
    insertionSort(arr1)
    for (i in arr1) {
        print("$i ")
    }
}

File: Algorithms/Insertion Sort/InsertionSort.go
package main

import (
	"fmt"
)
 
func main() {
 
	items := []int{78, 7, 2, 851, 33, 47, 58, 65}
	insertionsort(items)
    fmt.Println(items)
}
 
  
func insertionsort(items []int) {
    var n = len(items)
    for i := 1; i < n; i++ {
        j := i
        for j > 0 {
            if items[j-1] > items[j] {
                items[j-1], items[j] = items[j], items[j-1]
            }
            j = j - 1
        }
    }
}
